internal/provider/strato/strato_handler.go

```diff
@@ -0,0 +1,76 @@
+package strato
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/TimothyYe/godns/internal/settings"
+	"github.com/TimothyYe/godns/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// URL the API address for Strato.
+	URL = "https://%s:%s@dyndns.strato.com/nic/update?hostname=%s.%s&myip=%s"
+)
+
+// DNSProvider struct.
+type DNSProvider struct {
+	configuration *settings.Settings
+}
+
+// Init passes DNS settings and store it to the provider instance.
+func (provider *DNSProvider) Init(conf *settings.Settings) {
+	provider.configuration = conf
+}
+
+func (provider *DNSProvider) UpdateIP(domainName, subdomainName, ip string) error {
+	return provider.updateIP(domainName, subdomainName, ip)
+}
+
+// updateIP update subdomain with current IP.
+func (provider *DNSProvider) updateIP(domain, subDomain, currentIP string) error {
+	client := utils.GetHTTPClient(provider.configuration)
+	resp, err := client.Get(fmt.Sprintf(URL,
+		domain,
+		provider.configuration.Password,
+		subDomain,
+		domain,
+		currentIP))
+
+	if err != nil {
+		// handle error
+		log.Error("Failed to update sub domain:", subDomain)
+		return err
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Error(err)
+		}
+	}(resp.Body)
+
+	if err != nil {
+		log.Error("Err:", err.Error())
+		return err
+	}
+
+	body, _ := io.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("Update IP failed: %s", string(body))
+		return fmt.Errorf("Update IP failed: %s", string(body))
```

Comment:
Please use lower-case for the error message.

Reply:
Oh, I mean the line 64, not the line 63

Reply:
Sorry my bad. Edited the wrong line